app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :username, presence: true, length: { minimum: 5, maximum: 20 }
```

Comment:
Additional field validations should be present (email, rating, resources, etc.)

Reply:
Field validations for email, rating, and resources now added, only checking for presence at the moment ; if it can be improved let me know. Devise also provides some helpful validations for email and password out of the box.